### PR TITLE
feat/AB#75137-use-button-for-close-in-all

### PR DESCRIPTION
**Pull request by estelafs**
# Description
To keep the consistency I searched for modals and pages that used the close icon + "Cancel" button instead of a "Close" button, removed the close icon of the ui-dialog and updated the components that used it changing the "Cancel" buttons for "Close" buttons, since it does the same thing and a lot of components already used this.
Also replaced `(click)="onClose()"` for  the`uiDialogClose` directive when the onClose method only action was to close the modal

## Useful links
- Please insert link to ticket: https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/75137

## Type of change
- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?
Please see the screenshots below

## Screenshots

close
preferences

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by RenzoPrats — APPROVED** [review]
It's working fine, looks good to me!

**Comment by unai-reliefapp** [comment]
https://github.com/ReliefApplications/oort-frontend/blob/223d9f868cd05b539c8527e11a6e8fc80a361459/libs/safe/src/lib/components/email-preview/email-preview.component.html#L57-L59

This one should be changed with the Close text? @estelafs 

Also maybe we can use this ticket to set all the close buttons using the uiCloseDialog directive instead using a click function in order to keep the components overkill method free and same case for all buttons:

```
<ui-button (click)="onClose()">
      {{ 'common.close' | translate }}
 </ui-button>
```
to
```
<ui-button uiDialogClose>
      {{ 'common.close' | translate }}
 </ui-button>
```

And remove any related function in the component class .ts

What do you think?

**Review by unai-reliefapp — COMMENTED** [review]
Please take a look to my comment below the description

**Comment by estelafs** [comment]
@unai-reliefapp good point! Well remembered, thanks

File: apps/back-office/src/app/components/add-form-modal/add-form-modal.component.html [hunk 1]
```diff
@@ -107,7 +107,7 @@ <h1>{{ 'models.form.new' | translate }}</h1>
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">{{ 'common.close' | translate }}</ui-button>
+    <ui-button uiFormFieldDirective>{{ 'common.close' | translate }}</ui-button>
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/components/add-form-modal/add-form-modal.component.html]:
```suggestion
    <ui-button uiDialogClose>{{ 'common.close' | translate }}</ui-button>
```

File: apps/back-office/src/app/components/add-resource-modal/add-resource-modal.component.html [hunk 2]
```diff
@@ -17,7 +17,7 @@ <h1>{{ 'models.resource.create' | translate }}</h1>
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">{{ 'common.close' | translate }}</ui-button>
+    <ui-button uiFormFieldDirective>{{ 'common.close' | translate }}</ui-button>
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/components/add-resource-modal/add-resource-modal.component.html]:
```suggestion
    <ui-button uiDialogClose>{{ 'common.close' | translate }}</ui-button>
```

File: apps/back-office/src/app/dashboard/pages/api-configurations/components/add-api-configuration/add-api-configuration.component.html [hunk 3]
```diff
@@ -21,7 +21,7 @@ <h1>{{ 'models.apiConfiguration.new' | translate }}</h1>
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">{{ 'common.close' | translate }}</ui-button>
+    <ui-button uiFormFieldDirective>{{ 'common.close' | translate }}</ui-button>
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/dashboard/pages/api-configurations/components/add-api-configuration/add-api-configuration.component.html]:
```suggestion
    <ui-button uiDialogClose>{{ 'common.close' | translate }}</ui-button>
```

File: apps/back-office/src/app/dashboard/pages/applications/components/chose-role/chose-role.component.html [hunk 4]
```diff
@@ -20,7 +20,7 @@ <h1>
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">{{ 'common.close' | translate }}</ui-button>
+    <ui-button uiFormFieldDirective>{{ 'common.close' | translate }}</ui-button>
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/dashboard/pages/applications/components/chose-role/chose-role.component.html]:
```suggestion
    <ui-button uiDialogClose>{{ 'common.close' | translate }}</ui-button>
```

File: apps/back-office/src/app/dashboard/pages/reference-datas/add-reference-data/add-reference-data.component.html [hunk 5]
```diff
@@ -17,7 +17,7 @@ <h1>
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">{{ 'common.close' | translate }}</ui-button>
+    <ui-button uiFormFieldDirective>{{ 'common.close' | translate }}</ui-button>
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/dashboard/pages/reference-datas/add-reference-data/add-reference-data.component.html]:
```suggestion
    <ui-button uiDialogClose>{{ 'common.close' | translate }}</ui-button>
```

File: libs/safe/src/lib/components/access/edit-access/edit-access.component.html [hunk 6]
```diff
@@ -38,7 +38,7 @@ <h1>
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">{{ 'common.close' | translate }}</ui-button>
+    <ui-button uiFormFieldDirective>{{ 'common.close' | translate }}</ui-button>
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/access/edit-access/edit-access.component.html]:
```suggestion
    <ui-button uiDialogClose>{{ 'common.close' | translate }}</ui-button>
```

File: libs/safe/src/lib/components/choose-record-modal/choose-record-modal.component.html [hunk 7]
```diff
@@ -46,7 +46,7 @@ <h1>
     >
       {{ (isSearchActivated ? 'common.select' : 'common.search') | translate }}
     </ui-button>
-    <ui-button (click)="onClose()">{{ 'common.cancel' | translate }}</ui-button>
+    <ui-button uiFormFieldDirective>{{ 'common.cancel' | translate }}</ui-button>
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/choose-record-modal/choose-record-modal.component.html]:
```suggestion
    <ui-button uiDialogClose>{{ 'common.cancel' | translate }}</ui-button>
```

File: libs/safe/src/lib/components/convert-modal/convert-modal.component.html [hunk 8]
```diff
@@ -46,7 +46,7 @@ <h1>{{ data.title }}</h1>
     </div>
   </ng-container>
   <ng-container ngProjectAs="actions" *ngIf="!loading">
-    <ui-button (click)="onClose()">{{ 'common.close' | translate }}</ui-button>
+    <ui-button uiFormFieldDirective>{{ 'common.close' | translate }}</ui-button>
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/convert-modal/convert-modal.component.html]:
```suggestion
    <ui-button uiDialogClose>{{ 'common.close' | translate }}</ui-button>
```

File: libs/safe/src/lib/components/mapping/mapping-modal/mapping-modal.component.html [hunk 9]
```diff
@@ -46,7 +46,7 @@ <h1 *ngIf="!isNew">
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">
+    <ui-button uiFormFieldDirective>
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/mapping/mapping-modal/mapping-modal.component.html]:
```suggestion
    <ui-button uiDialogClose>
```

File: libs/safe/src/lib/components/record-modal/record-modal.component.html [hunk 10]
```diff
@@ -53,7 +53,7 @@
   </ng-container>
   <ng-container ngProjectAs="actions">
     <ng-container *ngIf="!loading && data.compareTo">
-      <ui-button (click)="onClose()">
+      <ui-button uiFormFieldDirective>
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/record-modal/record-modal.component.html]:
```suggestion
      <ui-button uiDialogClose>
```

File: libs/safe/src/lib/components/record-modal/record-modal.component.html [hunk 11]
```diff
@@ -66,7 +66,7 @@
       </ui-button>
     </ng-container>
     <ng-container *ngIf="!loading && !data.compareTo">
-      <ui-button (click)="onClose()" variant="danger">
+      <ui-button uiFormFieldDirective variant="danger">
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/record-modal/record-modal.component.html]:
```suggestion
      <ui-button uiDialogClose variant="danger">
```

File: libs/safe/src/lib/components/roles/components/add-role/add-role.component.html [hunk 12]
```diff
@@ -15,7 +15,7 @@ <h1>{{ title | translate }}</h1>
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">
+    <ui-button uiFormFieldDirective>
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/roles/components/add-role/add-role.component.html]:
```suggestion
    <ui-button uiDialogClose>
```

File: libs/safe/src/lib/components/users/components/add-user/add-user.component.html [hunk 13]
```diff
@@ -64,7 +64,7 @@ <h1>{{ 'components.users.invite.add' | translate }}</h1>
     </form>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">
+    <ui-button uiFormFieldDirective>
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/users/components/add-user/add-user.component.html]:
```suggestion
    <ui-button uiDialogClose>
```

File: libs/safe/src/lib/components/users/components/invite-users/invite-users.component.html [hunk 14]
```diff
@@ -104,7 +104,7 @@ <h1>{{ 'components.users.invite.confirm' | translate }}</h1>
     </kendo-grid>
   </ng-container>
   <ng-container ngProjectAs="actions">
-    <ui-button (click)="onClose()">
+    <ui-button uiFormFieldDirective>
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/users/components/invite-users/invite-users.component.html]:
```suggestion
    <ui-button uiDialogClose>
```

**Review by unai-reliefapp — CHANGES_REQUESTED** [review]

**Review by unai-reliefapp — APPROVED** [review]

**Review by MwanPygmay — COMMENTED** [review]
-Merged 2.x.x
-Removed the 'closable' parameter of the ui dialog component
-Added a close button in the bottom for all dialog components (some were missing)
-Added a button in the top right part of the dialogs (the div around the ng-content is just there so that if the header is missing, for example in the expanded widgets (maybe a bug btw), the close button is still to the right)

**Comment by AntoineRelief** [comment]
@MwanPygmay 
when you push new changes that affect the ui, please change the screenshots

File: libs/ui/src/lib/dialog/dialog.component.html [hunk 15]
```diff
@@ -1,16 +1,17 @@
...
-  icon="close"
-  [variant]="'danger'"
-  [uiTooltip]="'common.close' | translate"
-  (click)="onClose()"
->
-</ui-button>
 <div class="flex flex-col h-full overflow-hidden">
   <!-- Dialog header -->
-  <ng-content select="header"></ng-content>
+  <div class="flex justify-between">
```
Comment (AntoineRelief) [inline libs/ui/src/lib/dialog/dialog.component.html]: @MwanPygmay 
I don't like
for me it looks worse than before:
Capture d’écran 2023-10-09 à 11 35 29

**Pull request merged**